packages/tag/src/components/Tag.js

```diff
@@ -1,33 +1,56 @@
 import React from "react";
 import PropTypes from "prop-types";
 import values from "object.values";
+import { ThemeProvider } from "styled-components";
+import { Flex } from "@rebass/grid";
+
+import StyledTag, { StyledText } from "./styledComponent/StyledTag";
 
-import StyledTag from "./styledComponent/StyledTag";
 import tagTypes from "../constants/tagTypes";
 
 const Tag = props => {
-  const { text, onRemove, value } = props;
+  const { text, onRemove, value, icon, ...style } = props;
+
   return (
-    <StyledTag {...props}>
-      {text}
-      {onRemove && (
-        <div className="icon" onClick={() => onRemove(value)}>
-          <i className="wg-close" />
-        </div>
-      )}
-    </StyledTag>
+    <ThemeProvider theme={style}>
```

Comment:
We should wrap a `Tag` component with `ThemeProvider` outside the `Tag` component.
Check the docs: https://www.styled-components.com/docs/advanced#theming

We will style normally the `Tag` component. The difference is that we get the `type` from the `theme` provided. That is if no theme was provided, the component will render with the default style. If a theme is provided, we use the properties that come from the custom theme.